### PR TITLE
Added wishlist

### PR DESCRIPTION
**Pull request by Shah-Arsalan**
Added the Wishlist Page : 

1. The user can now click on the heart icon on the product card to add it to the wishlist 
2. After clicking on the same heart icon in the wishlist it gets removed from the wishlist

### CODE REVIEW

File: src/Components/Nav/Nav.jsx [hunk 1]
```diff
@@ -1,10 +1,12 @@
 import "./Nav.css";
 import { useNavigate } from "react-router-dom";
 import { useAuth } from "../../Contexts/auth-context";
+import { useData } from "../../Contexts/DataContext";
```
Comment (shrey27) [inline src/Components/Nav/Nav.jsx:5]: go for barrel exports, by using index.js file for common export of all the context functions and hooks

File: src/WebPages/WishListPage/Wishlist.css [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  .wishlist-product-container {
+    grid-template-columns: 1fr 1fr;
+  }
+}
+
+@media screen and (max-width: 600px) {
+  .wishlist-product-container {
+    grid-template-columns: 1fr;
+  }
+}
```
Comment (shrey27) [inline src/WebPages/WishListPage/Wishlist.css:54]: I would suggest using mobile first approach if you have to use media queries multiple times. 
UI can be easily responsive, with mobile first approach as it reduces dependency on media queries.
This also removes repetition from code.

**Review by shrey27 — APPROVED** [review]
Good to merge

File: src/App.js [hunk 3]
```diff
@@ -7,8 +7,13 @@ import { LandingPage } from "./WebPages/LandingPage/LandingPage";
 import { ProductListing } from "./WebPages/ProductListing/ProductListing";
 import { LoginPage } from "./WebPages/loginPage/LoginPage";
 import { useAuth } from "./Contexts/auth-context";
+import { Wishlist } from "./WebPages/WishListPage/Wishlist";
+import { useData } from "./Contexts/DataContext";
 
 function App() {
+  const { state } = useData();
+  console.log("wishlist", state.wishlist);
+  console.log("productdata", state.productdata);
```
Comment (anuj-kumary) [inline src/App.js]: Remove unwanted comments

File: src/Components/ProductCard/ProductCard.css [hunk 4]
```diff
@@ -48,3 +48,11 @@
 .vertical-card-component {
   height: 500px;
 }
+
+.card-skeleton .fav-icon {
+  color: black;
+}
+
+.card-skeleton .fas:hover {
+  color: red;
```
Comment (anuj-kumary) [inline src/Components/ProductCard/ProductCard.css]: Don't hardcode the color value use as a variable name

File: src/WebPages/ProductListing/ProductListing.css [hunk 5]
```diff
@@ -18,7 +18,6 @@ a {
 .card-skeleton .fas {
   padding: 0.5rem;
   background-color: rgb(238, 242, 247);
```
Comment (Rohanmond) [inline src/WebPages/ProductListing/ProductListing.css:20]: better to remove hardcoded value

File: src/WebPages/WishListPage/Wishlist.css [hunk 6]
```diff
@@ -0,0 +1,56 @@
+@import url(https://wanderer-store.netlify.app/main.css);
+@import url(https://wanderer-store.netlify.app//index.css);
+/* @import url(https:wanderer-store.netlify.app/WebPages/productListing/productlist.css); */
```
Comment (Rohanmond) [inline src/WebPages/WishListPage/Wishlist.css]: remove these commments

File: src/WebPages/WishListPage/Wishlist.jsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import { ProductCard } from "../../Components/ProductCard/ProductCard";
+import { useData } from "../../Contexts/DataContext";
+import "./Wishlist.css";
+const Wishlist = () => {
+  const { state } = useData();
+  console.log("inWishlist", state);
+  const wishlistData = state.productdata.filter(
+    (ele) => ele.inWishList === true
+  );
+  console.log("wishlistdata", wishlistData);
```
Comment (Rohanmond) [inline src/WebPages/WishListPage/Wishlist.jsx]: remove thses console

**Review by Rohanmond — APPROVED** [review]

File: src/WebPages/WishListPage/Wishlist.css (hunk 2 above)
Comment (Shah-Arsalan) [reply src/WebPages/WishListPage/Wishlist.css:54]: Thanks shrey for the comment ! I am planning to work on the responsiveness after completing the logic part . I hope that's fine

**Pull request merged**